tuf/data/types.go

```diff
@@ -17,6 +18,53 @@ import (
 	"github.com/docker/notary"
 )
 
+// GUN type for specifying gun
+type GUN string
+
+func (g GUN) String() string {
+	return string(g)
+}
+
+// NewGUN generates a GUN object from the provided string
+func NewGUN(g string) GUN {
```

Comment:
I'm not sure I see how `NewGun("gun")` is better than `GUN("gun")`, which you get for free